src/app/lab/components/dataset-toggle.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { Dataset } from '../../models/dataset.model';
+import { ProjectData } from '../../models/project-data.model';
+import { DatasetService } from '../../services/dataset.service';
+
+import * as _ from 'lodash';
+
+/*  Dataset Toggle Component
+
+    -- Requires project input
+    Expected use:
+        <ccl-dataset-toggle
+            [projectData]="your_project.project_data">
+*/
+
+@Component({
+  selector: 'ccl-dataset-toggle',
+  template: `
+    <div class="dropdown button-group">
+        <button class="button"
+               *ngFor="let s of datasets"
+               [disabled]="!checkValidDataset(s)"
+               [ngClass]="{'active': s.name === projectData.dataset.name}"
+               (click)="onDatasetSelected(s, $event)">
+            {{ s.name }}
+        </button>
+    </div>
+  `
+
+})
+export class DatasetToggleComponent implements OnInit {
+
+    @Input() projectData: ProjectData;
+    public datasets: Dataset[] = [];
+    private DEFAULT_DATASET_NAME = 'NEX-GDDP';
+
+    constructor(private datasetService: DatasetService) {}
+
+    ngOnInit() {
+        this.getDatasets();
+    }
+
+    public onDatasetSelected(dataset: Dataset, event?: Event) {
+        if (!this.isValidDataset(dataset)) {
+            this.selectDefaultDataset();
+        }
+        this.projectData.dataset = dataset;
+        this.projectData.models.forEach(model => {
+            model.enabled = _.includes(dataset.models, model.name);
+        });
+
+        if (event) {
+            event.preventDefault();
+        }
+    }
+
+    // set to default, or first valid option for the selected city
+    private selectDefaultDataset() {
+        let dataset = this.datasets.find(s => {
+            return s.name === this.DEFAULT_DATASET_NAME;
+        });
```

Comment:
these short functions can be simplified:
` let dataset = this.datasets.find(s => s.name === this.DEFAULT_DATASET_NAME);`